engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
Is there a need to cache host if nodeId is cached i.e. do we know `nodeId.getHost()` can be null later because of the same protocol buffer issue?